### PR TITLE
Fixes #9219: Searching on os type does not work

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/9219

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala [hunk 1]
```diff
@@ -276,6 +276,12 @@ object QSLdapBackend {
    * Mapping between attributes and their filter in the backend
    */
   object QSAttributeLdapFilter {
+
+    //the list of value to avoid in OC for matching
+    val ocRudderTypes = Set("top", OC_RUDDER_NODE, OC_POLICY_SERVER_NODE, OC_NODE
+                            , OC_RULE, OC_PARAMETER, OC_RUDDER_NODE_GROUP)
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: shouldn't you avoid also OC_SPECIAL_TARGET, OC_ACTIVE_TECHNIQUE_LIB_VERSION, OC_ABSTRACT_RULE_WITH_CF3POLICYDRAFT, OC_RULE_WITH_CF3POLICYDRAFT, OC_RULE_WITH_CF3POLICYDRAFT, OC_NODE_CONFIGURATION,OC_PROPERTY, or are they filtered differently ?

File: rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala [hunk 2]
```diff
@@ -404,25 +410,31 @@ object QSLdapBackend {
...
-          testers.find(f => f(token)).flatMap(findValue)
+          testers.find(f => f(token)).flatMap(findValue(a.getValues, _))
         }
 
         if(a.getName == A_OC) {
-          None
+          //here we must just avoid "top" and "kind" object class
+          //and also remove the "node" part for "linuxNode", etc
+          val values = a.getValues.flatMap(s =>
+            if(ocRudderTypes.contains(s)) None else Some(s.replace("Node", ""))
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: i'm not really fond of the twice hardcoded "Node" - could you at least use a constant ?

File: rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala (hunk 1 above)
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: We only select object in OC_NODE, OC_RUDDER_NODE, OC_RULES, OC_PARAMETER, OC_RUDDER_GROUP (and their children), so we don't need to filter these one.

File: rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala (hunk 2 above)
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: why not

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala (hunk 1 above)
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: ok, thanks for the clarification

**Pull request merged**